adagrams/game.py

```diff
@@ -1,11 +1,87 @@
+LETTER_POOL = {
+    'A': 9, 'B': 2, 'C': 2,'D': 4, 'E': 12, 'F': 2, 'G': 3, 'H': 2, 'I': 9, 
+    'J': 1, 'K': 1, 'L': 4, 'M': 2, 'N': 6, 'O': 8, 'P': 2,  'Q': 1, 'R': 6, 
+    'S': 4, 'T': 6, 'U': 4, 'V': 2, 'W': 2, 'X': 1, 'Y': 2, 'Z': 1
+}
+
 def draw_letters():
-    pass
+    letters_not_drawn = list(LETTER_POOL.keys())
+    letters_drawn = []
+    letter_count = 0
 
-def uses_available_letters(word, letter_bank):
-    pass
+    for letter in letters_not_drawn:
+        if letter not in letters_drawn and len(letters_drawn) < 10:
+            letters_drawn.append(letter)
+            letter_count += 1
+    return letters_drawn
 
+def uses_available_letters(word, letter_bank):
+    word = word.upper()
+    word_dict = {}
+    is_valid = False
+    letter_count = 0
+    for letter in word:
+        if letter in letter_bank and word.count(letter) <= letter_bank.count(letter):
```

Comment:
we are only just learning about Big O and a function's time and space complexity, but let's explore how we can make this more efficient. 

Right now we have a loop inside a loop (`if...in` has a loop under the hood). That makes the code O(n^2). Consider calculating a frequency map of the letters before entering the for loop on line 23. Building a frequency map is a linear operation aka O(n), but we could look up the letters in constant time aka O(1) with a frequency map (`if letter in dictionary` versus `if letter in list`).

That's very technical, but to summarize: using a dictionary to look up each letter in the word and subtracting from the dictionary would be more time efficient than checking if the letter is in the `letter_bank` list.